test/max-response-size.js

```diff
@@ -0,0 +1,103 @@
+'use strict'
+
+const { test } = require('tap')
+const { Client, errors } = require('..')
+const { createServer } = require('http')
+
+test('max response size', (t) => {
+  t.plan(4)
+
+  t.test('default max default size should allow all responses', (t) => {
+    t.plan(3)
+
+    const server = createServer()
+    t.teardown(server.close.bind(server))
+
+    server.on('request', (req, res) => {
+      res.end('hello')
+    })
+
+    server.listen(0, () => {
+      const client = new Client(`http://localhost:${server.address().port}`, { maxResponseSize: -1 })
+      t.teardown(client.close.bind(client))
+
+      client.request({ path: '/', method: 'GET' }, (err, { statusCode, body }) => {
+        t.error(err)
+        t.equal(statusCode, 200)
+        const bufs = []
+        body.on('data', (buf) => {
+          bufs.push(buf)
+        })
+        body.on('end', () => {
+          t.equal('hello', Buffer.concat(bufs).toString('utf8'))
+        })
+      })
+    })
+  })
+
+  t.test('max response size set to zero should allow only empty responses', (t) => {
+    t.plan(3)
+
+    const server = createServer()
+    t.teardown(server.close.bind(server))
+
+    server.on('request', (req, res) => {
+      res.end()
+    })
+
+    server.listen(0, () => {
+      const client = new Client(`http://localhost:${server.address().port}`, { maxResponseSize: 0 })
+      t.teardown(client.close.bind(client))
+
+      client.request({ path: '/', method: 'GET' }, (err, { statusCode, body }) => {
+        t.error(err)
+        t.equal(statusCode, 200)
+        const bufs = []
+        body.on('data', (buf) => {
+          bufs.push(buf)
+        })
+        body.on('end', () => {
+          t.equal('', Buffer.concat(bufs).toString('utf8'))
+        })
+      })
+    })
+  })
+
+  t.test('should throw an error if the response is too big', (t) => {
+    t.plan(3)
+
+    const server = createServer()
+    t.teardown(server.close.bind(server))
+
+    server.on('request', (req, res) => {
+      res.end('hello')
+    })
+
+    server.listen(0, () => {
+      const client = new Client(`http://localhost:${server.address().port}`, {
+        maxResponseSize: 1
+      })
+
+      t.teardown(client.close.bind(client))
+
+      client.request({ path: '/', method: 'GET' }, (err, { body }) => {
+        t.error(err)
+        body.on('error', (err) => {
+          t.ok(err)
+          t.type(err, errors.ResponseExceededMaxSizeError)
+        })
+      })
+    })
+  })
+
+  t.test('invalid max response size should throw an error', (t) => {
+    t.plan(2)
+
+    t.throws(() => {
+      Client('http://localhost:3000', { maxResponseSize: 'hello' })
```

Comment:
```suggestion
      new Client('http://localhost:3000', { maxResponseSize: 'hello' })
```

Reply:
In this way I got a `standard` error:

```bash
❯ npm run lint

> undici@5.11.0 lint
> standard | snazzy

standard: Use JavaScript Standard Style (https://standardjs.com)

/Users/mateonunez/dev/undici/test/max-response-size.js
   97:7  error  Do not use 'new' for side effects  no-new
  100:7  error  Do not use 'new' for side effects  no-new

✖ 2 problems
```

There is a way to avoid it without disabling the lint rule?

Reply:
Assign the object to a variable or use eslint-disable-line